File: THPT/2024/DE KSHSG L3, KIM LIÊN- TIN/DE KSHSG L3/CODE/PYTHON/CHIAKEO.py
def chia_keo():
    with open("CHIAKEO.INP", "r") as infile:
        # Đọc giá trị n và m từ dòng đầu tiên
        n, m = map(int, infile.readline().split())
        
        # Đọc tất cả các giá trị x trên cùng một dòng
        x_list = list(map(int, infile.readline().split()))

        mx = -1
        p = -1
        
        for i in range(1, n + 1):
            # Lấy giá trị x từ x_list
            x = x_list[i - 1]
            
            # Tính toán div = x // m + (x % m > 0)
            div = x // m + (x % m > 0)
            
            # Kiểm tra và cập nhật mx, p
            if mx <= div:
                mx = div
                p = i

    # Ghi kết quả ra tệp CHIAKEO.OUT
    with open("CHIAKEO.OUT", "w") as outfile:
        outfile.write(str(p))

# Gọi hàm để xử lý
chia_keo()

File: THPT/2024/DE KSHSG L3, KIM LIÊN- TIN/DE KSHSG L3/CODE/C++/CHIAKEO.cpp
#include <fstream>
using namespace std;

int main() {
    ifstream infile("CHIAKEO.INP");
    ofstream outfile("CHIAKEO.OUT");

    int n, m, mx = -1, p = -1;
    infile >> n >> m;

    for (int i = 1; i <= n; i++) {
        int x;
        infile >> x;
        int div = x / m + (x % m > 0);
        if (mx <= div) {
            mx = div;
            p = i;
        }
    }

    outfile << p;

    infile.close();
    outfile.close();
    return 0;
}

